sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.getAssessmentAwardDetails.e2e-spec.ts

```diff
@@ -0,0 +1,263 @@
+      disbursementSchedule: firstDisbursementSchedule,
+      student,
+      institutionLocation: collegeFLocation,
+    });
+    await db.disbursementReceipt.save(disbursementReceipt);
+
+    const awards = {};
+    const disbursementReceiptsValues = [] as DisbursementReceiptValue[];
+    firstDisbursementSchedule.disbursementValues.forEach((disbursement) => {
+      awards[`disbursement1${disbursement.valueCode.toLowerCase()}`] =
+        disbursement.valueAmount;
+      disbursementReceiptsValues.push(
+        createFakeDisbursementReceiptValue(
+          disbursement.valueCode,
+          disbursement.valueAmount,
+          { disbursementReceipt },
+        ),
+      );
+    });
+    await db.disbursementReceiptValue.save(disbursementReceiptsValues);
+
+    const finalAwards = {};
+    finalAwards[`disbursementReceipt1Id`] = disbursementReceipt.id;
+
+    disbursementReceiptsValues.forEach((disbursementReceiptsValue) => {
+      finalAwards[
+        `disbursementReceipt1${disbursementReceiptsValue.grantType.toLowerCase()}`
+      ] = disbursementReceiptsValue.grantAmount;
+    });
+
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/${assessment.id}/award`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        applicationNumber: application.applicationNumber,
+        applicationStatus: application.applicationStatus,
+        institutionName:
+          assessment.offering.educationProgram.institution.operatingName,
+        offeringIntensity: assessment.offering.offeringIntensity,
+        offeringStudyStartDate: getDateOnlyFormat(
+          assessment.offering.studyStartDate,
+        ),
+        offeringStudyEndDate: getDateOnlyFormat(
+          assessment.offering.studyEndDate,
+        ),
+        estimatedAward: {
+          disbursement1Date: getDateOnlyFormat(
+            firstDisbursementSchedule.disbursementDate,
+          ),
+          disbursement1Status:
+            firstDisbursementSchedule.disbursementScheduleStatus,
+          disbursement1COEStatus: firstDisbursementSchedule.coeStatus,
+          disbursement1MSFAANumber: MASKED_MSFAA_NUMBER,
+          disbursement1MSFAAId: firstDisbursementSchedule.msfaaNumber.id,
+          disbursement1MSFAACancelledDate:
+            firstDisbursementSchedule.msfaaNumber.cancelledDate,
+          disbursement1MSFAADateSigned:
+            firstDisbursementSchedule.msfaaNumber.dateSigned,
+          disbursement1TuitionRemittance:
+            firstDisbursementSchedule.tuitionRemittanceRequestedAmount,
+          disbursement1Id: firstDisbursementSchedule.id,
+          disbursement1DocumentNumber: firstDisbursementSchedule.documentNumber,
+          ...("disbursement1bcsl" in awards && {
```

Comment:
Since we have control of the data that is supposed to be generated, why do we need extra conditions? Maybe I am missing something. Can you please elaborate?

Reply:
yup, you are correct. I just did it to make it dynamic.  updating